File: 프로그래머스/lv2/장정윤/구명 보트.py
# https://school.programmers.co.kr/learn/courses/30/lessons/42885

def solution(people, limit):
    # 최소한의 구명보트 수 구하기, 최대 2명, 길이 제한
    answer = 0
    # 최소, 최대가 한팀
    left, right = 0, len(people)-1
    
    people.sort()
    
    while left <= right:
        if people[left] + people[right] <= limit:
            left += 1
        right -= 1
        answer += 1
        
    return answer

File: 프로그래머스/lv2/김원래/구명보트.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <list>

using namespace std;

int solution(vector<int> people, int limit) {
    int answer = 0;
    
    int begin = 0;
    int last = people.size() - 1;
    
    sort(people.begin(), people.end());
    
    while(begin <= last)
    {
        if(limit - people[last] >= people[begin])
        {
            begin++;
            last--;
        }
        else last--;
        
        ++answer;
    }
    
    
    return answer;
}

/*

구명보트는 2명 탈 수 있다.
구명보트는 무게 제한이 있다.

구명보트를 최대한 적게 사용하여 모든 사람을 구출하려고 한다.

1 2 4 5

3

*/